tests/unit/database.test.ts

```diff
@@ -0,0 +1,56 @@
+import knex, { Knex } from 'knex';
+import { mocked } from 'ts-jest/utils';
+import dateformat from 'dateformat';
+import StaffSchedule from '../../src/wms/Interfaces/StaffSchedule';
+import { Database } from '../../src/wms/Database';
+
+jest.mock('knex');
+const mknex = mocked(knex, true);
+const ent = {
+  site_id: 'string',
+  staff_id: 2,
+  status: 'string',
+  event_date: 'string',
+  event_start: 'string',
+  event_end: 'string',
+};
+const mKnex = ({
+  select: jest.fn().mockReturnThis(),
+  innerJoin: jest.fn().mockReturnThis(),
+  from: jest.fn().mockReturnThis(),
+  where: jest.fn(() => [ent]),
+  destroy: jest.fn().mockResolvedValue('destroyed'),
+} as unknown) as Knex;
+
+mknex.mockImplementation(
+// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  () => mKnex,
+);
+
+describe('Database calls', () => {
+  const database = new Database();
+  beforeEach(() => {
```

Comment:
remove if unused

Reply:
It is used. I'm not quite sure why it's tucked in there. I'll move it with the other declarations at a later date.